### PR TITLE
fix(annotations): animated fade trigger

### PR DESCRIPTION
**Pull request by nickofthyme**
## Summary

Adds `trigger` option to specify which triggers to act of thus allowing the user to opt-in to only the animations they desire for a given `Spec`.

BREAKING CHANGES: moves the animation configuration from the `AnnotationSpecStyles` to the `AnnotationSpec` and removes existing prop on the styles.

Screen Recording 2022-06-02 at 09 08 14 PM

## Details

Removed `animation` prop on `AnnotationSpecStyle`.

### Checklist

- [x] The proper **chart type** label has been added (e.g. `:xy`, `:partition`)
- [x] The proper **feature** labels have been added (e.g. `:interactions`, `:axis`)
- [ ] New public API exports have been added to `packages/charts/src/index.ts`
- [ ] Unit tests have been added or updated to match the most common scenarios
- [x] The proper documentation and/or storybook story has been added or updated

### CODE REVIEW

**Comment by nickofthyme** [comment]
@markov00 Still need to iron out a few kinks with the marker hover style but would still like to get your feedback, especially on naming and the API changes.

File: packages/charts/src/chart_types/xy_chart/utils/specs.ts [hunk 1]
```diff
@@ -836,6 +837,19 @@ export interface LineAnnotationDatum<D = any> {
...
+export const AnnotationAnimationTrigger = Object.freeze({
+  FadeOnFocusingOthers: 'FadeOnFocusingOthers' as const,
+});
+/** @public */
+export type AnnotationAnimationTrigger = $Values<typeof AnnotationAnimationTrigger>;
+
+/** @public */
+export interface AnnotationAnimations {
+  triggers?: AnnotationAnimationTrigger[];
+  options?: AnimationOptions;
```
Comment (markov00) [inline packages/charts/src/chart_types/xy_chart/utils/specs.ts:850]: I'm a bit concerned about this choice: the type name is `AnnotationAnimations` but we have a single `AnimationOptions` configuration that partially describes the animation (it describes basically duration and delay) but the same configurations are applied to multiple "triggers"
A trigger is a key that describes the actual animation and what should trigger it `fade` `on` `focusing others` and at this stage is fine for me, but I don't understand why we have single option per multiple triggers. If we have something like "blink on highlight" we probably want a different configuration, don't we?

I see this being rewritten as:

```ts
type LineAnnotationSpec {
animations?: Array<{trigger: Trigger, options: Options}>;
}
```

In this way, each trigger can have its own options set

Comment (nickofthyme) [reply packages/charts/src/chart_types/xy_chart/utils/specs.ts]: 691f6a5

File: packages/charts/src/utils/themes/theme.ts [hunk 2]
```diff
@@ -713,6 +713,8 @@ export interface LineAnnotationStyle {
   line: StrokeStyle & Opacity & Partial<StrokeDashArray>;
   /**
    * Animation options
+   *
+   * @deprecated use `animation.options` prop on annotation specs
```
[the remark this answers is not in the document]
Comment (nickofthyme) [reply packages/charts/src/utils/themes/theme.ts]: 691f6a5

File: packages/charts/src/chart_types/xy_chart/utils/specs.ts [hunk 3]
```diff
@@ -836,6 +837,19 @@ export interface LineAnnotationDatum<D = any> {
   header?: string;
 }
 
+/** @public */
+export const AnnotationAnimationTrigger = Object.freeze({
+  FadeOnFocusingOthers: 'FadeOnFocusingOthers' as const,
```
[the remark this answers is not in the document]
Comment (nickofthyme) [reply packages/charts/src/chart_types/xy_chart/utils/specs.ts:842]: Based on zoom chat I think many triggers can be linked to one animated key/value.

File: .gitignore [hunk 4]
```diff
@@ -5,6 +5,7 @@ test/spec/screenshots/diff/**/*.png
 test/spec/screenshots/screen/**/*.png
 test/failure-screenshots/**/*.png
 
+.env
```
Comment (markov00) [inline .gitignore:8]: not related to that change:
I'm seeing an issue where the first line marker always goes back to full opacity when moving the mouse over other elements

https://user-images.githubusercontent.com/1421091/171808778-6539e803-521b-491e-9a71-6f12f8ddc328.mov

Comment (nickofthyme) [reply .gitignore:8]: Are you talking about the crosshair? That is separate from the annotation.

Comment (nickofthyme) [reply .gitignore:8]: Strange, I can't reproduce this locally on Chrome. Are you still seeing this? If so what browser?

Screen Recording 2022-06-03 at 11 50 37 AM

Comment (markov00) [reply .gitignore:8]: I removed the cache and looks correct now, strange that's the first time I see a cache problem like that

File: packages/charts/src/chart_types/xy_chart/utils/specs.ts (hunk 1 above)
Comment (markov00) [reply packages/charts/src/chart_types/xy_chart/utils/specs.ts]: Looks good to me now! Thanks

File: packages/charts/src/utils/themes/theme.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (markov00) [reply packages/charts/src/utils/themes/theme.ts]: Perfect

**Review by markov00 — APPROVED** [review]
Changes looks good to me.
The issues with the line marker were solved by clearing the cache of the deployment page

**Comment by nickofthyme** [comment]
Confirmed these changes fix issues with `markerBody` mentioned in https://github.com/elastic/kibana/pull/131642#discussion_r881509220

Screen Recording 2022-06-06 at 07 30 44 PM

**Pull request merged**